Engine/Mixins/IMixin.cs

```diff
@@ -0,0 +1,122 @@
+using System;
+
+namespace OpenTap
+{
+    /// <summary>
+    /// Mixins can be used with EmbedPropertiesAttribute to give extra functionality to a test step.
+    /// </summary>
+    public interface IMixin : ITapPlugin
+    {
+        
+    }
+
+    abstract class MixinEvent<T2> where T2: IMixin
+    {
+        protected static T1 Invoke<T1>(object target, Action<T2, T1> f, T1 arg) 
+        {
+            var emb = TypeData.GetTypeData(target).GetBaseType<EmbeddedTypeData>();
+            if (emb == null) return arg;
+            foreach (var mem in emb.GetEmbeddingMembers())
+            {
+                if (!mem.TypeDescriptor.DescendsTo(typeof(T2))) continue;
+                if (mem.Readable == false) continue;
+                if (mem.GetValue(target) is T2 mixin)
+                {
+                    f(mixin, arg);
```

Comment:
What does it look like if `f` throws? It's probably obvious which mixin threw, and what the `MixinEvent` was, but I don't think the trace will have any information about what step or resource the mixin is attached to, or what the input was? 

Should we add something like a `MixinException` with some additional information, and an inner exception?

Reply:
We dont need MixinException for now.